### PR TITLE
Helm ownership transfer

### PR DESCRIPTION
**Pull request by jveski**
Adds test coverage for taking and giving up resource ownership with Helm, along with one small change to allow purposeful resource orphaning to enable this workflow.

### CODE REVIEW

File: internal/controllers/reconciliation/controller.go [hunk 1]
```diff
@@ -189,6 +189,9 @@ func (c *Controller) reconcileResource(ctx context.Context, prev, resource *reco
 		if current == nil || current.GetDeletionTimestamp() != nil {
 			return false, nil // already deleted - nothing to do
 		}
+		if comp.Annotations["eno.azure.io/deletion-strategy"] == "orphan" {
```
Comment (AYM1607) [inline internal/controllers/reconciliation/controller.go:192]: nit: Expose these as consts?

Comment (AYM1607) [reply internal/controllers/reconciliation/controller.go:192]: We might need to reference them from the ask side

**Review by AYM1607 — APPROVED** [review]
LGTM!

File: internal/controllers/reconciliation/controller.go (hunk 1 above)
Comment (jveski) [reply internal/controllers/reconciliation/controller.go:192]: I have weird opinions about consts lol. Let's chat sometime and hash out how to approach consts across the project :)

**Pull request merged**